### ISSUE TITLE
Something I've configured keeps deleting all my violentmonkey scripts upon browser restart.

### ISSUE THREAD

**Issue by Hyperi0n1337** [violentmonkey/violentmonkey#1875]
I think it's an about:config setting I've changed but I don't exactly know which. Any ideas what might be causing it? Whenever I restart nightly or when it has to restart to apply updates all my violentmonkey scripts get cleared. All my extensions however, are intact, like ublock etc.

Anyone know what might be the problem?

I get this error from violentmonkey when firefox starts

![image](https://github.com/violentmonkey/violentmonkey/assets/5292436/737e6283-7841-4786-8ceb-383b95e40d97)

**Comment by tophf** [comment]
1. Assuming you don't use an external utility like CCleaner, it may be `Always use private browsing mode` setting in Firefox.
2. Please click the text of the error and copy it here.

**Comment by tophf** [comment]
Yep, it didn't help me identify the cause. Try comparing modified properties in about:config of the two profiles. Anyway, I don't see what Violentmonkey can do about a setting in the browser.

**Comment by tophf** [comment]
BTW if there's a triangle at the beginning of the error in the background console, click it and see maybe there's more info.

**Comment by F9y4ng** [comment]
I've also had this problem and can reproduce it 100% on Firefox, but it doesn't affect normal  operation. (Reopening the menu can bring it back to normal)

## Reproduction Methods:
Opening Firefox and click the VM icon as fast as you can will 100% report the following error:

![error](https://github.com/violentmonkey/violentmonkey/assets/15855068/22e8db3e-c1f0-4693-bcfb-8e0538cce1eb)

## For example:
![test](https://github.com/violentmonkey/violentmonkey/assets/15855068/66379766-9898-4d5c-86f3-74fdf8088d8c)

**Comment by tophf** [comment]
Thanks, but the error in the popup is not what causes the original problem reported in this issue. It's a consequence of an error in the background context which you can see in about:debugging as I described in the previous comments.

**Comment by tophf** [comment]
Also, please try Violentmonkey beta.

**Comment by F9y4ng** [comment]
> Thanks, but the error in the popup is not what causes the original problem reported in this issue. It's a consequence of an error in the background context which you can see in about:debugging as I described in the previous comments.

I've seen the method you mentioned above.

However, I can't pre-open the VM console to get the error message with all Firefox processes closed, because the bug is only triggered when a Firefox process is started in a new process.

So I can only tell you the method that can reproduce this problem 100% and I don't know if it will help you.

**Comment by tophf** [comment]
If the error occurred in the background context you will see it even if the console wasn't opened previously.

**Comment by tophf** [comment]
I've fixed the popup error in ab6b7d5a. It won't solve the originally reported problem though.

**Comment by F9y4ng** [comment]
> If the error occurred in the background context you will see it even if the console wasn't opened previously.

no error report in the background context.

![PS_2023-08-25_21-25-39](https://github.com/violentmonkey/violentmonkey/assets/15855068/37904d57-8ac6-49a1-b0b0-e6c0846b991a)
